### PR TITLE
feat(button-group/chip/content-item): add theming refactoring

### PR DESCRIPTION
**Pull request by torgeadelin**
Add CSS variables to enable theming for webclient. (non-breaking changes) for following components
- typography (hyperlinks)
- list item
- button icon
- chip
- content-item
- input

The additions do not cover full component theming support but rather only the necessary bits to allow good ui/ux on the webclient. 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: core/scss/components/input/input.scss [hunk 1]
```diff
@@ -321,8 +321,10 @@
     &.#{$input__class}--filled {
       @include input-color(
         $background-color: $input__background--filled,
+        $background-color-css-var: --md-background-tertiary,
         $background-color-disabled: $input__background--filled,
-        $background-color-hover: $input__background--filled--hover
+        $background-color-hover: $input__background--filled--hover,
+        $background-color-hover-css-var: --md-background-quaternary,
```
Comment (InteractiveTimmy) [inline core/scss/components/input/input.scss:327]: Do we want to include `$background-color-disabled-css-var`? Not sure if this is a part of the mixin.

File: core/scss/ng/settings/settings.scss [hunk 2]
```diff
@@ -8101,7 +8101,8 @@ $blockquote-padding: rem-calc(9 20 0 19) !default;
 $blockquote-border-width: 4px !default;
 $blockquote-padding: 0.5rem 1rem calc(0.25rem + #{$blockquote-border-width})
   1rem !default;
-$blockquote-border: $blockquote-border-width solid $md-gray-50 !default;
+$blockquote-border: $blockquote-border-width solid !default;
+$blockquote-border-color: $md-gray-50 !default;
```
Comment (InteractiveTimmy) [inline core/scss/ng/settings/settings.scss:8105]: Assuming a neutral color works in this use case instead of a css var?

**Review by InteractiveTimmy — APPROVED** [review]
Looks good to me! Few questions though, nonblocking.

File: core/scss/components/input/input.scss (hunk 1 above)
Comment (torgeadelin) [reply core/scss/components/input/input.scss:327]: It's in the mixin as well.

File: core/scss/ng/settings/settings.scss (hunk 2 above)
Comment (torgeadelin) [reply core/scss/ng/settings/settings.scss:8105]: Yes.

**Review by InteractiveTimmy — APPROVED** [review]
LGTM!

**Pull request merged**